atomicapp/nulecule/__init__.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+import copy
+from atomicapp.utils import Utils
+
+
+class NuleculeBase(object):
+    def load(self):
+        pass
+
+    def load_config(self, config, params, namespace):
+        for param in params:
+            value = config.get(namespace).get(param['name']) or \
+                config.get('general').get(param['name']) or \
+                param.get('default')
+            if value is None:
+                value = Utils.askFor(param['name'], param)
+            config[namespace][param['name']] = value
+        return config
+
+    def load_params(self, params):
+        pass
+
+    def run(self, provider):
+        raise NotImplementedError
+
+    def stop(self, provider):
+        raise NotImplementedError
+
+    def install(self):
+        raise NotImplementedError
+
+    def uninstall(self):
+        raise NotImplementedError
+
+
+class Nulecule(NuleculeBase):
+    """
+    This represents an application compliant with Nulecule specification.
+    A Nulecule instance can have instances of Nulecule and Nulecule as
+    components. A Nulecule instance knows everything about itself and its
+    componenents, but does not have access to it's parent's scope.
+    """
+    def __init__(self, id, specversion, metadata, graph, requirements=None,
+                 params=None, config=None, namespace='general'):
+        self.id = id
+        self.specversion = specversion
+        self.metadata = metadata
+        self.graph = graph
+        self.requirements = requirements
+        self.params = params or []
+        self.namespace = namespace
+        self.config = self.load_config(config, self.params, self.namespace)
```

Comment:
What about moving this to `load()`, or moving  `load_components()` out of `load` ?